mtags/src/main/scala-3.2/scala/meta/internal/pc/MetalsSealedDesc.scala

```diff
@@ -7,5 +7,6 @@ import dotty.tools.dotc.core.Symbols.Symbol
 object MetalsSealedDesc:
   def sealedStrictDescendants(sym: Symbol)(using Context): List[Symbol] =
     sym.sealedStrictDescendants.filterNot(child =>
-      child.is(Sealed) && (child.is(Abstract) || child.is(Trait))
+      (child.is(Sealed) && (child.is(Abstract) || child.is(Trait)))
+        || child.isPrivate
```

Comment:
How will this behave with regards to a symbol that is non-private in the current scope?

```scala
object Test:
   private enum A:
      case B, C
   def testMe(a: A) = a match@@
```

Reply:
Here we would get exhaustive match completion, because we check only if the children i.e. B and C are private.

Although you are right, there is a problem in case like
```scala
sealed trait A
abstract class B extends A
private abstract class C extends A {
  val a: A = ???
  a match@@ // `case C` should be visible, but it's not
}
```

Reply:
Maybe `(child.isPublic || child.isAccessibleFrom(sym.info))` ?

Reply:
It works, thanks!

Reply:
Cool! Could you also add a test with @keynmol 's case?

Reply:
> Cool! Could you also add a test with @keynmol 's case?

You mean this one?

```scala
private enum A:
      case B, C
   def testMe(a: A) = a match@@
```

Reply:
Yes!